write_test.go

```diff
@@ -667,3 +668,49 @@ func setUndefinedLength(e *Element) *Element {
 	e.ValueLength = tag.VLUndefinedLength
 	return e
 }
+
+// TestWriteElement tests that Write and Writer.WriteElement produces identical results.
```

Comment:
perhaps a better test may be something similar to the Write test now--that is lets write out a Dataset using Writer.WriteElement, then read it back in with Parser and check if they're identical? 

I mention this because L686-L707 here is making an internal assumption about what Write does, when maybe we shouldn't in order to test the behavior of WriteElement. e.g. what if Write changes in the future to add a special metadata element in all things written out by the parser? It would seem weird to have to modify the test for something meant to test WriteElement in that case. 

This is somewhat true in testing it against Parse, except we know Parse should always produce a consistent dataset with whatever stream of elements was written with WriteElement.

I guess this is a minor point, so please let me know what you think.

Reply:
We can also address this in a separate PR if we want, but the other comment/discussion is the only main blocker to merging imo.